src/main/java/roomescape/config/LoginMemberArgumentResolver.java

```diff
@@ -0,0 +1,54 @@
+package roomescape.config;
+
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.core.MethodParameter;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import roomescape.domain.Member;
+import roomescape.service.MemberService;
+
+@Component
+public class LoginMemberArgumentResolver implements HandlerMethodArgumentResolver {
+
+    private final MemberService memberService;
+
+    public LoginMemberArgumentResolver(final MemberService memberService) {
+        this.memberService = memberService;
+    }
+
+    @Override
+    public boolean supportsParameter(final MethodParameter parameter) {
+        return parameter.getParameterType().equals(Member.class);
+    }
+
+    @Override
+    public Object resolveArgument(final MethodParameter parameter, final ModelAndViewContainer mavContainer,
+                                          final NativeWebRequest webRequest, final WebDataBinderFactory binderFactory)
+            throws Exception {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) webRequest.getNativeRequest();
+        String token = extractTokenFromCookie(httpServletRequest.getCookies());
+
+        Long memberId = Long.valueOf(Jwts.parserBuilder()
+                .setSigningKey(Keys.hmacShaKeyFor("Yn2kjibddFAWtnPJ2AFlL8WXmohJMCvigQggaEypa5E=".getBytes()))
+                .build()
+                .parseClaimsJws(token)
+                .getBody().getSubject());
+
+        return memberService.findMemberById(memberId);
+    }
+
+    private String extractTokenFromCookie(final Cookie[] cookies) {
+        for (Cookie cookie : cookies) {
+            if (cookie.getName().equals("token")) {
```

Comment:
매직 넘버를 상수로 관리해보면 어떨까요? 그리고 해당 라인에서 혹시나 발생할 수 있는 NPE를 방지하기 위해선 어떻게 코드를 수정하면 될까요?

Reply:
```suggestion
            if (TOKEN_COOKIE_NAME.equals(cookie.getName())) {
```
상수와 NPE 방지를 반영하여 수정해보았습니다!